accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java

```diff
@@ -1651,6 +1654,16 @@ public Edge getEdge(String edgeId, EnumSet<FetchHint> fetchHints, Long endTime,
 
             final long timerStartTime = System.currentTimeMillis();
             ScannerBase scanner = createBatchScanner(getDataTableName(), ranges, new org.apache.accumulo.core.security.Authorizations());
+            if (timestamp != null) {
```

Comment:
This change is a bit scary and I'm not sure how to verify unless it's tested in a real world scenario. I can image that it could be possible that a SPV was written in the past and now that the timestamp is not in the key I need to look it up based on the property's timestamp, which could be off? I'm not really sure.